src/core/friendly_errors/validate_params.js

```diff
@@ -344,9 +344,9 @@ if (typeof IS_MINIFIED !== 'undefined') {
   const isNumber = param => {
     switch (typeof param) {
       case 'number':
-        return true;
+        return isFinite(param);
       case 'string':
-        return !isNaN(param);
+        return !(isNaN(param) || param == "null");
```

Comment:
There is another special case that needs to be accounted for here which is when `param === ""` or `param === " "` for any number of spaces in the string. Accounting for this can potentially add some complexity, alternatively we can attempt to do `parseFloat(param)` first to directly test if the result is `NaN` which also have the benefit of possibly not needing to use different switch cases here.